go-controller/pkg/util/multi_network.go

```diff
@@ -761,6 +761,61 @@ func ValidateNetConf(nadName string, netconf *ovncnitypes.NetConf) error {
 		return fmt.Errorf("the subnet attribute must be defined for layer2 primary user defined networks")
 	}
 
+	if netconf.Topology != types.LocalnetTopology && netconf.Name != types.DefaultNetworkName {
```

Comment:
so this means we will do it for UDPNs and UDSNs as well which is good.

Reply:
my only question would be whether this should happen in validate OR parse portions because it seems like we do parseCIDR here already but then perhaps we do it again at ParseNAD but I guess its fine.